doc/port-si/Port_SI_Per_Speed.md

```diff
@@ -0,0 +1,230 @@
+</br></br>
+
+
+
+# Changes to support the enhancements
+
+
+1. Changes in SfpStateUpdateTask thread:</br></br>
+    With the port SI per speed enhancements applied, we rely on the lane speed when we lookup in _media_settings,json_. Hence, this flow has to be triggered not only by insertion/removal of modules, but by speed configuration changes as well.</br>
+    In order to establish the dependency of port SI configurations on lane speed, we need to be able to monitor speed configuration changes. Therefore, we will add to the SfpStateUpdateTask a listener to detect such changes: a new member will be added to SfpStateUpdateTask to listen to changes in STATE_DB.PORT_TABLE, and once such change is detected, _notify_media_settings()_ will be trigerred. Additionally, the SfpStateUpdateTask thread will have a new dictionary that will store the speed and number of lanes for each port.
+</br></br>
+
+
+2. The _XCVRD::Notifty_media_settings()_ function should be modified to support the updated format of _media_settings_json_:
+
+   - The method _get_media_settings_key()_ should be extended:
+
+      We need to extend the key used for lookup in the '_media_settings.json_' file to consider lane speed. </br>
+      Currently, there are two types of keys: 'vendor_key' (vendor name + vendor part number, for example: 'AMPHENOL-1234') and 'media_key' (media type + media_compliance_code + media length, for example: 'QSFP28-40GBASE-CR4-1M'). </br>
+      In the new format of '_media_settings.json_', the '_get_media_settings_key()_' method will return three values instead of the two values described above. The additional value returned from this method will be the 'lane_speed_key', for example: '400GAUI-8' (where '400' refers to the port speed and '8' refers to the lane count). </br></br>
+
+      How will the 'lane_speed_key' be calculated? </br>
+      Each module contains a list called Application Advertisements in its EEPROM, which is a list of all speeds the module is compatible with. For example:
+      
+      ```
+      Application Advertisement: 
+        400GAUI-8 C2M (Annex 120E) - Host Assign (0x1) - Active Cable assembly with BER < 2.6x10^-4 - Media Assign (0x1)
+        IB EDR (Arch.Spec.Vol.2) - Host Assign (0x11) - Active Cable assembly with BER < 10^-12 - Media Assign (0x11)
+        200GAUI-4 C2M (Annex 120E) - Host Assign (0x11) - Active Cable assembly with BER < 2.6x10^-4 - Media Assign (0x11)
+        CAUI-4 C2M (Annex 83E) without FEC - Host Assign (0x11) - Active Cable assembly with BER < 10^-12 - Media Assign (0x11)
+      ```
+
+      We will use this list to derive the lane_speed_key. We will iterate over this list and return the item whose port speed and lane count match the port speed and lane count for the corresponding port that were extracted from STATE_DB into the newly added dictionary (The mapping between ports and their speeds and lane counts as described earlier).
```

Comment:
@tshalvi the source of truth for port Speed and lane mapping is config DB not STATE_DB

Reply:
Certainly, I made sure to update the HLD with that detail.